_pytest/_code/code.py

```diff
@@ -354,7 +354,10 @@ def __init__(self, tup=None, exprinfo=None):
             if exprinfo is None and isinstance(tup[1], AssertionError):
                 exprinfo = getattr(tup[1], 'msg', None)
                 if exprinfo is None:
-                    exprinfo = str(tup[1])
+                    try:
+                        exprinfo = str(tup[1])
+                    except UnicodeEncodeError:
+                        exprinfo = unicode(tup[1])
```

Comment:
Looking at `tup[1]`, it always seems to contain an unicode object, even when there are byte strings in the assertion:

```python
-> exprinfo = str(tup[1])
(Pdb) tup[1]
AssertionError(u"assert 'a' == 'b'\n  - a\n  + b",)
```

So I think the proper fix would be to always use `unicode()` and do the check with `u'assert '` below.